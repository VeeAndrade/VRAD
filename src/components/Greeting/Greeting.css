.user-info {
  width: 100%;
  border: solid 2px orange
}