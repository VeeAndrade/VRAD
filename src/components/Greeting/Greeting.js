import React from 'react';
import { NavLink } from 'react-router-dom';
import './Greeting.css';

const Greeting = ({ user, purpose, userFavorites }) => {
  if(!user || !purpose) {
    return ''
  } else {
    return(
      <article className='user-info'>
        <section className='heading-container-wrapper'>
          <NavLink className='signout-btn' to='/'>Signout</NavLink>
          <NavLink to='/favorites'>{userFavorites.length}</NavLink>
        </section>
        <p>
          Hello, {user}. You've chosen {purpose} as your
          reason for being here!
        </p>
      </article>
    )}

  }

export default Greeting;
