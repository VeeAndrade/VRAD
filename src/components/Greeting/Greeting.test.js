import React from 'react';
import  { shallow } from 'enzyme';
import Greeting from './Greeting';

describe('Greeting', () => {
  it('should match the snapshot with all the data passed in correctly', () => {
    const mockFavorites = {
      listing_id: 3,
      area_id: 590,
      name: "Hip RiNo Party Spot",
      address: {
        street: "2250 Lawrence St",
        zip: "80205"
      },
      details: {
        neighborhood_id: 5124122,
        superhost: true,
        seller_source: "91jss1",
        beds: 3,
        baths: 2.5,
        cost_per_night: 420,
        features: [
          "hot tub",
          "espresso machine"
        ],
      },
      dev_id: "u4gh2j",
      area: "rino",
      db_connect: 834470
    }
    const wrapper = shallow(<Greeting 
      user='VEE'
      purpose='Vacation'
      userFavorites={[mockFavorites]}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})