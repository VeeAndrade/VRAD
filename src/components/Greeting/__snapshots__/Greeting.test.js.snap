// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting should match the snapshot with all the data passed in correctly 1`] = `
<article
  className="user-info"
>
  <section
    className="heading-container-wrapper"
  >
    <NavLink
      className="signout-btn"
      to="/"
    >
      Signout
    </NavLink>
    <NavLink
      to="/favorites"
    >
      1
    </NavLink>
  </section>
  <p>
    Hello, 
    VEE
    . You've chosen 
    Vacation
     as your reason for being here!
  </p>
</article>
`;
